### PR TITLE
Optimized PlainHasher hashing. Trie insertions are >15% faster

### PR DESCRIPTION
**Pull request by debris**
`PlainHasher::write` before:

```
test write ... bench:      18,025 ns/iter (+/- 1,425)
```

`PlainHasher::write` after:

```
test write ... bench:          58 ns/iter (+/- 10)
```

before:

```
test trie_insertions_32_mir_1k      ... bench:   3,956,331 ns/iter (+/- 577,529)
test trie_insertions_32_ran_1k      ... bench:   4,009,316 ns/iter (+/- 567,403)
test trie_insertions_random_mid     ... bench:   2,900,765 ns/iter (+/- 283,349)
test trie_insertions_six_high       ... bench:   2,981,152 ns/iter (+/- 331,667)
test trie_insertions_six_low        ... bench:   5,828,811 ns/iter (+/- 593,180)
test trie_insertions_six_mid        ... bench:   3,889,366 ns/iter (+/- 110,769)
test trie_iter                      ... bench:   6,755,724 ns/iter (+/- 1,199,892)
```

after:

```
test trie_insertions_32_mir_1k      ... bench:   3,262,684 ns/iter (+/- 118,147)
test trie_insertions_32_ran_1k      ... bench:   3,290,951 ns/iter (+/- 78,756)
test trie_insertions_random_mid     ... bench:   2,654,520 ns/iter (+/- 744,629)
test trie_insertions_six_high       ... bench:   2,492,217 ns/iter (+/- 236,359)
test trie_insertions_six_low        ... bench:   4,989,133 ns/iter (+/- 86,834)
test trie_insertions_six_mid        ... bench:   3,314,200 ns/iter (+/- 84,884)
test trie_iter                      ... bench:   6,749,411 ns/iter (+/- 6,490,631)
```

### CODE REVIEW

File: util/benches/trie.rs [hunk 1]
```diff
@@ -45,14 +45,14 @@ fn random_bytes(min_count: usize, diff_count: usize, seed: &mut H256) -> Vec<u8>
...
-	seed[0..r].into_vec()
+	seed[0..r].to_vec()
 }
 
 fn random_value(seed: &mut H256) -> Bytes {
 	*seed = seed.sha3();
 	match seed[0] % 2 {
 		1 => vec![seed[31];1],
-		_ => seed.into_vec(),
+		_ => seed.to_vec(),
```
Comment (debris) [inline util/benches/trie.rs]: unrelated benchmark fix

**Comment by debris** [comment]
I moved `PlainHasher` to a separate crate. here is the [source](https://github.com/debris/plain_hasher/blob/master/src/lib.rs)

**Comment by rphmeier** [comment]
indentation wrong here: https://github.com/debris/plain_hasher/blob/master/src/lib.rs#L45
BTW, shouldn't it be a `paritytech` crate -- @arkpar and I are also authors of PlainHasher

**Comment by debris** [comment]
@rphmeier updated readme and moved the repo to https://github.com/paritytech/plain_hasher

I will also add core-devs as owners on crates.io once they fix this bug: https://github.com/rust-lang/crates.io/issues/997

**Comment by arkpar** [comment]
I don't really see why `PlainHasher` needs to be on crates.io. It is too parity-specific to be generally useful.

**Comment by debris** [comment]
@arkpar fixed
